src/Kiota.Builder/Plugins/PluginsGenerationService.cs

```diff
@@ -127,22 +145,56 @@ private PluginManifestDocument GetManifestDocument(string openApiDocumentPath)
             Functions = [.. functions.OrderBy(static x => x.Name, StringComparer.OrdinalIgnoreCase)]
         };
     }
-    private (OpenApiRuntime[], Function[]) GetRuntimesAndFunctionsFromTree(OpenApiUrlTreeNode currentNode, string openApiDocumentPath)
+
+    private static OpenApiManifestInfo ExtractInfoFromDocument(OpenApiInfo? openApiInfo)
+    {
+        var manifestInfo = new OpenApiManifestInfo();
+
+        if (openApiInfo is null)
+            return manifestInfo;
+
+        string? descriptionForModel = null;
+        string? legalUrl = null;
+        string? logoUrl = null;
+        string? privacyUrl = null;
+        string contactEmail = string.IsNullOrEmpty(openApiInfo.Contact?.Email)
+            ? DefaultContactEmail
+            : openApiInfo.Contact.Email;
+
+        if (openApiInfo.Extensions.TryGetValue(OpenApiDescriptionForModelExtension.Name, out var descriptionExtension) &&
+            descriptionExtension is OpenApiDescriptionForModelExtension extension &&
+            !string.IsNullOrEmpty(extension.Description))
+            descriptionForModel = extension.Description.CleanupXMLString();
+        if (openApiInfo.Extensions.TryGetValue(OpenApiLegalInfoUrlExtension.Name, out var legalExtension) && legalExtension is OpenApiLegalInfoUrlExtension legal)
+            legalUrl = legal.Legal;
+        if (openApiInfo.Extensions.TryGetValue(OpenApiLogoExtension.Name, out var logoExtension) && logoExtension is OpenApiLogoExtension logo)
+            logoUrl = logo.Url;
+        if (openApiInfo.Extensions.TryGetValue(OpenApiPrivacyPolicyUrlExtension.Name, out var privacyExtension) && privacyExtension is OpenApiPrivacyPolicyUrlExtension privacy)
+            privacyUrl = privacy.Privacy;
+
+        return new OpenApiManifestInfo(descriptionForModel, legalUrl, logoUrl, privacyUrl, contactEmail);
+
+    }
+    private const string DefaultContactName = "publisher-name";
+    private const string DefaultContactEmail = "publisher-email@example.com";
+    private sealed record OpenApiManifestInfo(string? DescriptionForModel = null, string? LegalUrl = null, string? LogoUrl = null, string? PrivacyUrl = null, string ContactEmail = DefaultContactEmail);
+    private static (OpenApiRuntime[], Function[]) GetRuntimesAndFunctionsFromTree(OpenApiUrlTreeNode currentNode, string openApiDocumentPath)
     {
         var runtimes = new List<OpenApiRuntime>();
         var functions = new List<Function>();
         if (currentNode.PathItems.TryGetValue(Constants.DefaultOpenApiLabel, out var pathItem))
         {
             foreach (var operation in pathItem.Operations.Values.Where(static x => !string.IsNullOrEmpty(x.OperationId)))
             {
+                var operationId = operation.OperationId.Replace('.', '_');
```

Comment:
if we do that we also need to set the operation id on the document so it matches

Reply:
Thanks for catching this. I'll probably change this back for now so that we can look into it separately as I see this leading to a document with different ids than the original which is a scenario which we may not want.

Reply:
We'll follow this up via https://github.com/microsoft/kiota/issues/4642 as it seems related.